File: backend/src/main/java/com/tourneyhub/backend/dto/match/MatchCreateDto.java
package com.tourneyhub.backend.dto.match;

import jakarta.validation.constraints.FutureOrPresent;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
public class MatchCreateDto {

    @NotNull
    private Long stageId;

    @NotNull
    private boolean teams;

    @NotNull
    @Size(min = 1, max = 8)
    private String code;

    @NotNull
    @FutureOrPresent
    private Date time;

    @NotNull
    private Long player1Id;

    @NotNull
    private Long player2Id;

    private Long refereeId;

    private Long streamerId;

    @NotNull
    private List<Long> commentatorIds;
}

File: tourneyhub/src/dto/schedule/match/MatchCreateDto.ts
import { Dayjs } from 'dayjs';

export interface MatchCreateDto {
    stageId: string,
    teams: boolean,
    code: string,
    time: Dayjs,
    player1Id: string,
    player2Id: string,
    refereeId: string,
    streamerId: string,
    commentatorIds: string[]
}